src/CSnakes.Runtime/PyObjectTypeConverter.cs

```diff
@@ -31,11 +35,6 @@ public static object PyObjectToManagedType(PyObject pyObject, Type destinationTy
             return ConvertToList(pyObject, destinationType);
         }
 
-        if (CPythonAPI.IsBuffer(pyObject) && destinationType.IsAssignableTo(bufferType))
```

Comment:
@tonybaloney I've moved this to `PyObject.As(Type type)`. See [my other comment for why](https://github.com/tonybaloney/CSnakes/pull/692/files#r2358004640).